app/services/targets/update_visibility_service.rb

```diff
@@ -26,21 +26,27 @@ def execute
     private
 
     def detach_from_prerequisites
-      TargetPrerequisite.transaction do
+      AssignmentPrerequisite.transaction do
         Targets::DetachFromPrerequisitesService.new([@target]).execute
       end
     end
 
     def remove_as_prerequisite
-      TargetPrerequisite.where(prerequisite_target: @target).destroy_all
+      AssignmentPrerequisite.where(
+        prerequisite_assignment: assignment
+      ).delete_all
     end
 
     def clear_milestone_settings
-      @target.update!(milestone: false, milestone_number: nil)
+      assignment.update!(milestone: false, milestone_number: nil) if assignment
```

Comment:
The `archived` field in the assignments table is set to true when a user creates a target with an assignment, and then later chooses No to the question "Does this target have an assignment?" This keeps the assignment data intact, so that if they later choose to answer Yes we can give back all their assignment data. 
I was thinking when a target is archived or unarchived, we don't touch the above behaviour of an assignment. Since we always get assignments from live targets this wouldn't cause any issues. Does that make sense?